### PR TITLE
[IMP] mail: cleanup naming in attachmentList model

### PR DESCRIPTION
**Pull request by phenix-factory**

### CODE REVIEW

**Comment by phenix-factory** [comment]
@seb-odoo Rename PR, has you asked it in an other.

File: addons/mail/static/src/models/attachment_image/attachment_image.js [hunk 1]
```diff
@@ -64,7 +64,7 @@ registerModel({
             if (this.attachmentList.attachmentBoxViewOwner) {
                 return 160;
             }
-            if (this.attachmentList.message) {
+            if (this.attachmentList.messageViewOwner.message) {
```
Comment (seb-odoo) [inline addons/mail/static/src/models/attachment_image/attachment_image.js:67]: `message` is required on `messageViewOwner` so we can drop the `.message`

**Comment by seb-odoo** [comment]
commit title a bit too long

File: addons/mail/static/src/models/attachment_list/attachment_list.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ import { clear, insertAndReplace, replace } from '@mail/model/model_field_comman
 
 registerModel({
     name: 'AttachmentList',
-    identifyingFields: [['composerView', 'messageView', 'attachmentBoxViewOwner']],
+    identifyingFields: [['composerView', 'messageViewOwner', 'attachmentBoxViewOwner']],
```
Comment (seb-odoo) [inline addons/mail/static/src/models/attachment_list/attachment_list.js:9]: Note: to finish it this clean, we should also make the same rename to `composerViewOwner` in another PR.

Comment (phenix-factory) [reply addons/mail/static/src/models/attachment_list/attachment_list.js]: This doesn't create much diff, so I included it inside this PR.

Comment (seb-odoo) [reply addons/mail/static/src/models/attachment_list/attachment_list.js]: Well it does make the PR a bit harder to review than the 2 separate parts, but ok now that it's done.

**Review by seb-odoo — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**